code/src/containers/products-container/ProductsContainer.tsx

```diff
@@ -44,31 +79,47 @@ const ProductsContainer = ({
     );
   }
 
-  const skeletonCards = repeatComponent(<ProductSkeleton />, loadingItemsCount);
-
-  const handleAddToCart = async (product: Product) => {
-    try {
-      if (user?.id) {
-        await addToCart({ productId: product.id, userId: user.id }).unwrap();
-        openDrawer(<CartDrawer />);
+  const handleCartIconClick = async (product: HandleCartIconClickParam) => {
+    if (user?.id) {
+      try {
+        if (!product.isInCart) {
+          await addToCart({
+            productId: product.id,
+            userId: user.id
+          }).unwrap();
+        } else {
+          openDrawer(<CartDrawer />);
+        }
+      } catch {
+        openSnackbarWithTimeout({
+          variant: "error",
+          messageTranslationKey: "cart.itemAddition.fail"
+        });
       }
-    } catch {
-      openSnackbarWithTimeout({
-        variant: "error",
-        messageTranslationKey: "cart.itemAddition.fail"
-      });
+    } else {
+      openModal(<AuthModal />);
```

Comment:
Try to use guard clause here. In my point of view it will make our code more readable

Here is an article to recollect this concept: https://medium.com/@timothydan/javascript-guard-clauses-64b999e3240

Reply:
Thank you, fixed